language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/FieldAccessCompletionResolver.java

```diff
@@ -288,4 +290,11 @@ private boolean withValidAccessModifiers(Node exprNode, MethodSymbol methodSymbo
         return isPublic || (!isPrivate && objModuleId.moduleName().equals(currentModule.moduleName())
                 && objModuleId.orgName().equals(currentPackage.packageOrg().value()));
     }
+
+    private String unescape(String name) {
+        if (name.startsWith(IDENTIFIER_LITERAL_PREFIX)) {
+            return IdentifierUtils.unescapeUnicodeCodepoints(name.substring(1));
+        }
+        return name;
+    }
```

Comment:
Why do we need to handle this specifically here? Shouldn't the value be escaped/ unescaped before consuming in LS?

Reply:
Since this is a syntax tree visitor, it's getting the values as-is. It's up to the visitor to decide whether to unescape or not right?

Reply:
My concern is, that we have some information preserved in the syntax tree and those information is not available in the semantic models. If we are going to have a similar implementation in other places, we do have to do the unescaping.

Reply:
Yeah.

Talked with @lochana-chathura, It is okay to have this `unescape()` function inside `IdentifierToken` and retrieve the unescaped form of the text of the name. So we can use it any other place we have a similar problem. WDYT?

Reply:
What's the possibility of preserving the actual value (with the quote) in the symbol as well? This is because if we get the  symbol after escaping or not, the symbol which is being visible to the will have the name without the quote

Reply:
Storing the unescaped actual value shouldn't be an issue I think. When @sanjana was working on the quoted identifier issue with symbols, we followed the approach James mentioned here: https://github.com/ballerina-platform/ballerina-lang/issues/28567. The issue with this approach is this only covers adding the quote back to reserved words. We can check the feasibility of preserving the original value the others and see.

Reply:
IMHO if we address the quoted identifier issue, this issue will be resolved automatically and will not be needing a custom fix.